src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
Yes you’re correct that would fail.  What I don’t understand is the reference to /^\d/

Because basically /\d/ checks for any digits in the string.

Can you send me what you think the if statement should be?


The only thing I can think to do is to check to see if the first character is a digit.  If it is then deep linking won’t work.  If not we can use the parse option.

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Thursday, December 3, 2015 at 12:45 PM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(formly-field): issue number 566 - Breaking change in 7.3.2 (#569)


In src/directives/formly-field.js:
>        // If we are working with a number then $parse wont work, default back to the old way for now
> -      if (angular.isNumber(key)) {
> -        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
> +      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
Hmmm, I believe that my example demonstrated the issue pretty well. I guess, here's a test. We need to make this pass :-)

it('should work with nested keys with numbers in the key', () => {
  const key = 'foo3bar.baz4foobar'
  const defaultValue = 'baz'
  scope.fields = [
    {template: input, key, defaultValue}
  ]
  scope.model = {}

  compileAndDigest()
  expect(scope.model.foo3bar.baz4foobar).to.equal(defaultValue)
})
Does that make sense? I think what we have now would actually cause this test to fail.

—
Reply to this email directly or view it on GitHub.